runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/fn/control/ProcessRemoteBundleOperation.java

```diff
@@ -100,6 +197,75 @@ public void finish() throws Exception {
       } catch (Exception e) {
         throw new RuntimeException("Failed to finish remote bundle", e);
       }
+
+      // TODO(BEAM-6274): do we have to put this in the "start" method as well?
+      try (RemoteBundle bundle =
```

Comment:
From my understanding so far, there should be only one call of getBundle in this operation. Why do you want to get another bundle in the process call?

Reply:
Like we talked yesterday, timers are allowed to A) fire other timers, and B) fire themselves. To implement this, we send the timer to the SDK Harness, wait for any elements, then check if any additional timers were fired. This is enough for now, but it'll be good to check the performance once more people are using it.